### PR TITLE
Fix: `is_update` when switching `no_staking`

### PR DESCRIPTION
**Pull request by OjusWiZard**
## Proposed changes

- [x] When changing `no_staking` to `staking` and vice-versa, the service must be updated.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: operate/services/manage.py [hunk 1]
```diff
@@ -748,9 +750,10 @@ def _deploy_service_onchain_from_safe(  # pylint: disable=too-many-statements,to
                 # TODO This has to be removed for Optimus (needs to be properly implemented). Needs to be put back for Trader!
                 or current_agent_bond != staking_params["min_staking_deposit"]
                 or on_chain_description != service.description
+                or (current_staking_program is None)
+                ^ (user_params.staking_program_id == NO_STAKING_PROGRAM_ID)
```
Comment (jmoreira-valory) [inline operate/services/manage.py:754]: Very difficult to understand what you mean with a XOR here. Can you elaborate?

Comment (OjusWiZard) [reply operate/services/manage.py:754]: here it means that if it was no_staking currently and want to stake or is already staked and want to run in no_staking. Then it should update

File: operate/services/manage.py [hunk 2]
```diff
@@ -551,23 +551,34 @@ def _deploy_service_onchain_from_safe(  # pylint: disable=too-many-statements,to
...
+        current_staking_params = sftxb.get_staking_params(
+            fallback_params=fallback_params,
+            staking_contract=get_staking_contract(
+                chain=ledger_config.chain,
+                staking_program_id=current_staking_program,
+            ),
+        )
+        if user_params.staking_program_id == current_staking_program:
+            target_staking_params = current_staking_program
+        else:
```
Comment (jmoreira-valory) [inline operate/services/manage.py:574]: Why this `if`?  If it's an optimization to avoid calling staking methods, the `get_staking_params` method already has an internal cache for staking parameters.

Comment (OjusWiZard) [reply operate/services/manage.py:574]: oh yeah, I did not see it

File: operate/services/manage.py [hunk 3]
```diff
@@ -719,48 +732,50 @@ def _deploy_service_onchain_from_safe(  # pylint: disable=too-many-statements,to
...
         # TODO Handle this in a more graceful way.
         agent_id = (
-            staking_params["agent_ids"][0]
-            if staking_params["agent_ids"]
+            target_staking_params["agent_ids"][0]
+            if target_staking_params["agent_ids"]
             else user_params.agent_id
         )
-        staking_params["agent_ids"] = [agent_id]
+        target_staking_params["agent_ids"] = [agent_id]
```
Comment (jmoreira-valory) [inline operate/services/manage.py:739]: I think this can be simplified, no?
target_staking_params already equals the fallback params if no staking.
`agent_id = target_staking_params["agent_ids"][0]`

**Review by jmoreira-valory — APPROVED** [review]

File: operate/services/manage.py (hunk 3 above)
Comment (OjusWiZard) [reply operate/services/manage.py:736]: yeah, this can be removed probably. But I'll not touch it for now because we could be missing some edge cases where `user_params.agent_id` is different from the `target_staking_params["agent_ids"][0]`. There is no test coverage also for this
